File: src/JSTest/Communication/Payloads/TestRunEventArgs.cs
using Microsoft.VisualStudio.TestPlatform.ObjectModel;
using System;
using System.Collections.Generic;
using System.Text;

namespace JSTest.Communication.Payloads
{
    [Serializable]
    public class TestCaseStartEventArgs : EventArgs
    {
        public TestCase TestCase;
    }

    [Serializable]
    public class TestCaseFoundEventArgs : TestCaseStartEventArgs
    {
    }

    [Serializable]
    public class TestCaseEndEventArgs : EventArgs
    {
        public TestResult TestResult;
    }
}

File: src/JSTest.Runner/ObjectModel/Payloads/TestRunEventArgs.ts
import { TestCase } from '../Common';
import { TestResult } from '..';

export interface TestCaseStartEventArgs {
    TestCase: TestCase;
}

export interface TestCaseEndEventArgs {
    TestResult: TestResult;
}

export interface TestCaseFoundEventArgs {
    TestCase: TestCase;
}